geode-core/src/integrationTest/java/org/apache/geode/internal/cache/DiskStoreImplIntegrationTest.java

```diff
@@ -100,6 +103,56 @@ public void queueSizeStatIncrementedAfterAsyncFlush() throws Exception {
     await().until(() -> diskStoreStats.getQueueSize() == 0);
   }
 
+  @Test
+  public void givenDiskStoreIsCreated_thenGetDiskUsagePercentageShouldBeZero() throws Exception {
+
+    File[] diskDirs = new File[2];
+    diskDirs[0] = temporaryDirectory.newFolder("dir1");
+    diskDirs[1] = temporaryDirectory.newFolder("dir2");
+
+    int[] diskDirSizes = new int[2];
+    diskDirSizes[0] = 20;
+    diskDirSizes[1] = 20;
+
+    cache.createDiskStoreFactory().setMaxOplogSize(2).setDiskDirsAndSizes(diskDirs, diskDirSizes)
+        .create(DISK_STORE_NAME);
+    DiskStore diskStore = cache.findDiskStore(DISK_STORE_NAME);
+
+    assertEquals(0, ((DiskStoreImpl) diskStore).getDiskUsagePercentage());
+  }
+
+  @Test
+  public void givenDiskStoreIsCreated_whenDataIsStored_thenGetDiskUsagePercentageShouldBeIncreased()
+      throws Exception {
+
+    final int NUM_ENTRIES = 100000;
+    final int ESTIMATED_USAGE_PERCENTAGE = 7; // Estimated disk percentage for NUM_ENTRIES
+    final int ALLOWED_MARGIN = 2;
+
+    File[] diskDirs = new File[2];
+    diskDirs[0] = temporaryDirectory.newFolder("dir1");
+    diskDirs[1] = temporaryDirectory.newFolder("dir2");
+
+    int[] diskDirSizes = new int[2];
+    diskDirSizes[0] = 20;
+    diskDirSizes[1] = 20;
+
+    cache.createDiskStoreFactory().setMaxOplogSize(2).setDiskDirsAndSizes(diskDirs, diskDirSizes)
+        .create(DISK_STORE_NAME);
+    Region region = cache.<String, String>createRegionFactory(RegionShortcut.PARTITION_PERSISTENT)
+        .setDiskStoreName(DISK_STORE_NAME).create(REGION_NAME);
+    DiskStore diskStore = cache.findDiskStore(DISK_STORE_NAME);
+
+    for (int i = 0; i < NUM_ENTRIES; i++) {
+      region.put(i, i);
+    }
+
+    assertNotEquals(0, ((DiskStoreImpl) diskStore).getDiskUsagePercentage());
+    assertTrue("Disk usage percentage is not in the expected margin.",
+        (((DiskStoreImpl) diskStore).getDiskUsagePercentage()
+            - ESTIMATED_USAGE_PERCENTAGE) <= ALLOWED_MARGIN);
+  }
+
```

Comment:
There are private helper methods in this class that do some similar setup as is being done here. Consider modifying the helper methods so that the added tests can use them, or inlining the helper methods since they are only used in one or two tests.

Reply:
When changing the percentage to float, this line changed, so not sure if the comment applies. Now the assert receives three parameters:
```
assertEquals(ESTIMATED_USAGE_PERCENTAGE, ((DiskStoreImpl) diskStore).getDiskUsagePercentage(), ALLOWED_MARGIN);
```

I also overloaded the putEntries method to reuse code:

```
  private void putEntries(int numToPut) {
    putEntries(aRegion,numToPut);
  }
  
  private void putEntries(Region region, int numToPut) {
    for (int i = 1; i <= numToPut; i++) {
      region.put(i, i);
    }
  }
```